pkg/controller/component/kbagent.go

```diff
@@ -278,47 +307,61 @@ func customExecActionImageNContainer(synthesizedComp *SynthesizedComponent) (str
 		actions = append(actions, &synthesizedComp.LifecycleActions.RoleProbe.Action)
 	}
 
-	var image, container string
+	// TODO: Discussion is needed for the definition of action name
+	actionNameMap := map[*appsv1alpha1.Action]string{
+		synthesizedComp.LifecycleActions.PostProvision:     "postpr",
+		synthesizedComp.LifecycleActions.PreTerminate:      "preter",
+		synthesizedComp.LifecycleActions.Switchover:        "switch",
+		synthesizedComp.LifecycleActions.MemberJoin:        "mbrin",
+		synthesizedComp.LifecycleActions.MemberLeave:       "mbrlv",
+		synthesizedComp.LifecycleActions.Readonly:          "readol",
+		synthesizedComp.LifecycleActions.Readwrite:         "readwr",
+		synthesizedComp.LifecycleActions.DataDump:          "datadp",
+		synthesizedComp.LifecycleActions.DataLoad:          "datald",
+		synthesizedComp.LifecycleActions.Reconfigure:       "reconf",
+		synthesizedComp.LifecycleActions.AccountProvision:  "accpr",
+		&synthesizedComp.LifecycleActions.RoleProbe.Action: "rolepb",
+	}
+
+	var images []string
+	var containerNames []string
+	var actionNames []string
 	for _, action := range actions {
 		if action == nil || action.Exec == nil {
 			continue
 		}
-		if action.Exec.Image != "" {
-			if len(image) > 0 && image != action.Exec.Image {
-				return "", nil, fmt.Errorf("only one exec image is allowed in lifecycle actions")
-			}
-			image = action.Exec.Image
+		image := action.Exec.Image
+		containerName := action.Exec.Container
+		if len(image) == 0 && len(containerName) == 0 {
+			continue
 		}
-		if action.Exec.Container != "" {
-			if len(container) > 0 && container != action.Exec.Container {
-				return "", nil, fmt.Errorf("only one exec container is allowed in lifecycle actions")
+		images = append(images, image)
+		containerNames = append(containerNames, containerName)
+		actionNames = append(actionNames, actionNameMap[action])
+	}
+
+	containers := make([]*corev1.Container, len(images))
+	for i := 0; i < len(images); i++ {
+		var c *corev1.Container
+		if len(containerNames[i]) > 0 {
+			for j, cc := range synthesizedComp.PodSpec.Containers {
+				if cc.Name == containerNames[i] {
+					c = &synthesizedComp.PodSpec.Containers[j]
+					break
+				}
 			}
-			container = action.Exec.Container
-		}
-	}
-
-	var c *corev1.Container
-	if len(container) > 0 {
-		for i, cc := range synthesizedComp.PodSpec.Containers {
-			if cc.Name == container {
-				c = &synthesizedComp.PodSpec.Containers[i]
-				break
+			if c == nil {
+				return nil, nil, nil, fmt.Errorf("exec container %s not found", containerNames[i])
 			}
+			if len(images[i]) > 0 && c.Image != images[i] {
```

Comment:
according to the api-doc, Exec.image and Exec.container 're mutually exclusive, only one of them should be provided.